protocol-units/da-sequencer/node/src/batch.rs

```diff
@@ -1,24 +1,95 @@
 use crate::error::DaSequencerError;
-use aptos_crypto::ed25519::{Ed25519PublicKey, Ed25519Signature};
+use bcs;
+use ed25519_dalek::{Signature, Verifier, VerifyingKey};
+use serde::{Deserialize, Serialize};
 
 #[derive(Debug)]
 pub struct RawData {
 	pub bytes: Vec<u8>,
 }
 
-#[derive(Debug)]
-pub struct FullnodeTx {}
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct FullnodeTx {
+	pub id: u64,
+	pub data: String,
+}
 
 #[derive(Debug)]
 pub struct DaBatch<Data> {
-	data: Data,
-	signature: Ed25519Signature,
-	signer: Ed25519PublicKey,
+	pub data: Data,
+	pub signature: Signature,
+	pub signer: VerifyingKey,
 }
 
-/// Batch write blobs.
-fn validate_batch(
+pub fn validate_batch(
 	new_batch: DaBatch<RawData>,
-) -> std::result::Result<DaBatch<FullnodeTx>, DaSequencerError> {
-	todo!()
+) -> Result<DaBatch<Vec<FullnodeTx>>, DaSequencerError> {
+	verify_batch_signature(&new_batch.data.bytes, &new_batch.signature, &new_batch.signer)?;
+
+	let txs: Vec<FullnodeTx> = bcs::from_bytes(&new_batch.data.bytes)
```

Comment:
See PR: #1155 and how grpc data are [deserialized](https://github.com/movementlabsxyz/movement/blob/6443612af3ba6f53faf66c87db1b5888351076c3/protocol-units/da-sequencer/node/src/server.rs#L76).

`data_bytes` is a Vec of Transactions.